openff/interchange/interop/gromacs/export/_export.py

```diff
@@ -222,7 +222,38 @@ def _write_atoms(
         top.write("[ atoms ]\n")
         top.write(";index, atom type, resnum, resname, name, cgnr, charge, mass\n")
 
-        for atom in molecule_type.atoms:
+        def _adjust_charges(charges: numpy.array, tolerance=8) -> numpy.array:
+            """
+            Adjust charges so that written charge for a molecule type is 0.
+            """
+            rounded_charges = numpy.round(charges, tolerance)
+            total_charge = numpy.round(numpy.sum(charges_to_write), 0)
+
+            def _rounding_error(_arr, _sum, _tolerance):
+                return numpy.round(numpy.sum(_arr) - _sum, _tolerance)
+
+            rounding_error = _rounding_error(rounded_charges, total_charge, tolerance)
+
+            if rounding_error != 0:
+                rounded_charges += numpy.round(
+                    -rounding_error / len(charges_to_write),
+                    tolerance,
+                )
```

Comment:
(blocking) Wouldn't 0 charges be affected here, contradicting the "never adjust 0 charges" comment below?

Reply:
You are right, should be resolved now.